### PR TITLE
5695 add keycloak authentication

### PR DESCRIPTION
**Pull request by tschia**

### CODE REVIEW

**Comment by tschia** [comment]
This change includes Dave's initial commit. 
To just view my changes you can view the individual commit:
https://github.com/bcgov/moh-ldap-account-transfer/pull/6/commits/9d863381487f02a9253d792c0c17740bc67b9cbe

File: backend/pom.xml [hunk 1]
```diff
@@ -47,6 +52,11 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.security</groupId>
+			<artifactId>spring-security-oauth2-jose</artifactId>
+			<version>5.5.2</version>
```
Comment (daveb-hni) [inline backend/pom.xml:58]: This overrides the manage version, why is this required?

File: backend/src/main/java/ca/bc/gov/hlth/accounttransfer/security/AudienceValidator.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+@Component
+public class AudienceValidator implements OAuth2TokenValidator<Jwt> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AudienceValidator.class);
+
+    private static final OAuth2Error error = new OAuth2Error(OAuth2ErrorCodes.INVALID_REQUEST, "The " + AUD + " claim is not valid",
+            "https://tools.ietf.org/html/rfc6750#section-3.1");
+
+
+    private static final String AUDIENCE = "LDAP-ACCOUNT-TRANSFER-API";
```
Comment (daveb-hni) [inline backend/src/main/java/ca/bc/gov/hlth/accounttransfer/security/AudienceValidator.java:29]: Why is this not specified in spring security props, ...jwt.audience?

**Review by daveb-hni — COMMENTED** [review]
Please see comments

File: backend/src/main/java/ca/bc/gov/hlth/accounttransfer/security/AudienceValidator.java (hunk 2 above)
Comment (tschia) [reply backend/src/main/java/ca/bc/gov/hlth/accounttransfer/security/AudienceValidator.java]: Just harded coded it in initial setup and forgot to move it. 
Fixed

File: backend/pom.xml (hunk 1 above)
Comment (tschia) [reply backend/pom.xml]: Let my IDE import, didn't notice it pulled a specific version. 
Fixed

**Review by daveb-hni — APPROVED** [review]
Changes look good.
